internal/db/group.go

```diff
@@ -103,8 +103,12 @@ func (d *Database) ListGroups(ctx context.Context, limit, offset int, match stri
 		db = db.Where("name LIKE ? OR uuid LIKE ?", "%"+match+"%", "%"+match+"%")
 	}
 	db = db.Order("name asc")
-	db = db.Limit(limit)
```

Comment:
An interesting question:
- I think this would've been okay because the limit and offset should be validated before they get to this function, but nothing wrong about defensive programming. But then should we also set an upper limit here or trust an upper limit was applied before reaching here?

Reply:
the way it was written, limit would be set to 0 (the default value) and NO records would be returned. Unless we intentionally set a limit that was higher than 0.

Reply:
i think that's because from the rebac endpoint they set the `pageSize`, and if they set `pageSize` to 0 that' fair to return 0 elements

Reply:
not all clients (e.g. jimm) use pagination

Reply:
I'd imagine max limits should be per query, not an overall max. No?

Reply:
for reference: juju returns everything when you do `Model`. As far as I've seen, there is no pagination.

https://github.com/juju/juju/blob/e3263a5bfcb34ede12f8a1601b9847b4d881a051/state/modeluser.go#L169-L243

Reply:
@kian99 with this dependency update those jimmctl tests that were sending limit=0 started failing, because nothing was returned. If you want all entries to be returned you should be setting limit=-1

Reply:
Ahh those tests started failing, didn't realise. Then this change is perfect imo.